### PR TITLE
Use `mapList()`

### PR DESCRIPTION
**Pull request by mernst**
1. Do you think this is a code improvement that should be merged?
2. Can you suggest how to fix the type-checking errors in the `typecheck` CI jobs?

### CODE REVIEW

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java [hunk 1]
```diff
@@ -2909,18 +2909,18 @@ public Node visitNewArray(NewArrayTree tree, Void p) {
...
+        List<Node> initializerNodes;
         if (initializers != null) {
-            for (ExpressionTree init : initializers) {
-                initializerNodes.add(assignConvert(scan(init, p), elemType));
-            }
+            initializerNodes =
+                    SystemUtil.mapList(
+                            init -> assignConvert(scan(init, p), elemType), initializers);
+        } else {
+            initializerNodes = new ArrayList<>();
```
Comment (mernst) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java:2923]: Using `Collections.emptyList()` is more efficient.

File: dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpressionConverter.java [hunk 2]
```diff
@@ -34,8 +34,13 @@ public JavaExpression convert(JavaExpression javaExpr) {
      * @return the list of converted expressions
      */
     public List<@PolyNull JavaExpression> convert(List<@PolyNull JavaExpression> list) {
-        return CollectionsPlume.mapList(
-                (JavaExpression expression) -> expression == null ? null : convert(expression),
+        return SystemUtil.mapList(
+                (@PolyNull JavaExpression expression) -> {
+                    if (expression == null) {
```
Comment (mernst) [inline dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpressionConverter.java:39]: I'm curious why a ternary operator does not work here.

File: framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java [hunk 3]
```diff
@@ -223,24 +223,18 @@ private static void fixUpRawTypes(
...
-                newTypeArgs.add(subTypeArgs.get(mapping.first).deepCopy());
-            }
+            List<? extends AnnotatedTypeMirror> subTypeArgs = declaredSubtype.getTypeArguments();
+            List<AnnotatedTypeMirror> newTypeArgs =
+                    SystemUtil.mapList(
+                            mapping -> subTypeArgs.get(mapping.first).deepCopy(),
+                            orderedByDestination);
+            declaredAsSuper.setTypeArguments(newTypeArgs);
+        } else {
+            declaredAsSuper.setTypeArguments(new ArrayList<>(0));
```
Comment (mernst) [inline framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java:236]: `Collections.emptyList()` is more efficient here.

File: javacutil/src/main/java/org/checkerframework/javacutil/SystemUtil.java [hunk 4]
```diff
@@ -244,8 +246,10 @@ public static String conjunction(String conjunction, List<?> elements) {
      * @param c a collection
      * @return a list of the results of applying {@code f} to the elements of {@code list}
      */
-    public static <FROM, TO> List<TO> mapList(
-            Function<? super FROM, ? extends TO> f, Collection<FROM> c) {
+    public static <
+                    @KeyForBottom FROM extends @UnknownKeyFor Object,
+                    @KeyForBottom TO extends @UnknownKeyFor Object>
+            List<TO> mapList(Function<? super FROM, ? extends TO> f, Collection<FROM> c) {
         return c.stream().map(f).collect(Collectors.toList());
```
Comment (mernst) [inline javacutil/src/main/java/org/checkerframework/javacutil/SystemUtil.java:253]: This is not the implementation in plume-util.  (That implementation changed and I thought you were going to pick up the new implementation.)

File: dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpressionConverter.java (hunk 2 above)
Comment (smillst) [reply dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpressionConverter.java:39]: https://github.com/typetools/checker-framework/issues/1170.

Comment (mernst) [reply dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpressionConverter.java:39]: Thanks.  I would add a reference to it.  (That will also allow us to clean this up when the issue is fixed.)

**Review by smillst — APPROVED** [review]
The typeCheck job should now pass, but if it fails again, I'll fix it.

**Pull request merged**